### DISCUSSION TITLE
use an environment variable to determine which application's logs to collect

### DISCUSSION THREAD

**Discussion by loveyang2012** [vectordotdev/vector#20722] [General]
In a Kubernetes environment, there are many application logs, and it seems that Vector collects all of them. However, I only want to collect the logs of my own application. Can I use an environment variable to determine which application's logs to collect? For example, when I start an Nginx application, I configure an environment variable during startup, and Vector detects this environment variable to collect Nginx logs.

**Comment by jszwedko** [comment]
There are a few options available on the `kubernetes_logs` source to filter the collected containers. For example: https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#extra_label_selector

**Reply by loveyang2012** [reply]
thinks
